engines/wasmtime/wasmtime.go

```diff
@@ -75,10 +77,22 @@ type (
 var _ = (wapc.Module)((*Module)(nil))
 var _ = (wapc.Instance)((*Instance)(nil))
 
-var engineInstance = engine{}
+type EngineOption func(e *engine)
+
+// WithEngine allows you to override the default engine. Defaults to
+// `wasmer.NewEngine`.
+func WithEngine(engineFn func() *wasmtime.Engine) EngineOption {
```

Comment:
Just curious: Why pass in a function to create the engine instead of passing in an engine instance?

Reply:
@pkedy Actually I don't see any reason other than an example I worked from may have used a creator func rather than the engine itself.  Is it better with passing the engine or just different as I think I can easily (famous last words) rework.

Reply:
note if we pass the engine instead, it becomes the same as existing wazero code which passes the runtime

```go
// EngineWithRuntime allows you to customize or return an alternative to
// DefaultRuntime,
func EngineWithRuntime(newRuntime NewRuntime) wapc.Engine {
	return &engine{newRuntime: newRuntime}
}
```

except wazero is missing the unwrap functions added here to the other two..

Reply:
one thing that may come up later is customization of instantiation of modules within a runtime-specific way beyond the portable `wapc.ModuleConfig`, but in that case one way out is to change it from a struct to an interface, in any case we can solve it when we get to it.